### PR TITLE
[Brave News] Fix bug where direct feeds didn't show up as followed

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Follow up for https://github.com/brave/brave-core/pull/15963, which fixed it's original issue but broke the button toggle state.

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. With #brave-news-subscribe-button enabled
2. Navigate to theatlantic.com
3. Follow `Best of the Atlantic`
4. The button state should update

### CODE REVIEW

File: browser/brave_news/brave_news_tab_helper.cc [hunk 1]
```diff
@@ -71,8 +71,10 @@ BraveNewsTabHelper::GetAvailableFeeds() {
 bool BraveNewsTabHelper::IsSubscribed(const FeedDetails& feed_details) {
   auto* publisher = controller_->publisher_controller()->GetPublisherForFeed(
       feed_details.feed_url);
-  return publisher && publisher->user_enabled_status ==
-                          brave_news::mojom::UserEnabled::ENABLED;
+  return publisher &&
```
Comment (simonhong) [inline browser/brave_news/brave_news_tab_helper.cc:74]: How about adding comments like below for example? I'm always confusing :)  
```
if (!publisher)
  return false;

// User chosen feed. It doesn't have enabled/disabled state.
// When it exists, it means subscribed. If not, publisher conttroller doesn't have it.
if (publisher->type == brave_news::mojom::PublisherType::DIRECT_SOURCE)
  return true;

// User can enable/disable publishers from default-provided feeds.
return publisher->user_enabled_status == brave_news::mojom::UserEnabled::ENABLED;
```

**Review by simonhong — APPROVED** [review]
++

File: browser/brave_news/brave_news_tab_helper.cc (hunk 1 above)
Comment (fallaciousreasoning) [reply browser/brave_news/brave_news_tab_helper.cc:74]: Good call!

**Comment by stephendonner** [comment]
While initially trying to test this for uplift, I encountered https://github.com/brave/brave-browser/issues/27052, just FYI for now.

**Comment by stephendonner** [comment]
## Verified `PASSED` using 

Brave | 1.48.15 Chromium: 108.0.5359.62 (Official Build) nightly (x86_64)
-- | --
Revision | 041930a89a990cfab0315a2d9f20d6429a4a67cf-refs/branch-heads/5359@{#938}
OS | macOS Version 11.7.1 (Build 20G918)

### Steps:
1. installed `1.48.15`
2. launched Brave
3. opened `brave://flags`
4. confirmed no `Brave News button` flag exists
5. loaded `theatlantic.com`
6. clicked on the `RSS`-feed icon in the URL bar
7. clicked on `Best of the Atlantic` 's `Follow` button
8. waited


#### Confirmed the `Follow` / `Unfollow` button state is accurately reflected

`brave://flags` | `theatlantic.com` | `Follow`  | `Unfollow`
---------------|------------------|------------|-----------
Screen Shot 2022-11-29 at 5 11 49 PM | Screen Shot 2022-11-29 at 5 11 59 PM | Screen Shot 2022-11-29 at 5 12 04 PM | Screen Shot 2022-11-29 at 5 12 14 PM


NOTE: logged https://github.com/brave/brave-browser/issues/27052 while testing

**Comment by fallaciousreasoning** [comment]
**Note:** I think https://github.com/brave/brave-browser/issues/27052 was introduced by https://github.com/brave/brave-core/pull/16038, so the fix for that will need an uplift too.

**Pull request merged**